guide/students/student_marks_management_service.bal

```diff
@@ -0,0 +1,115 @@
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerina/mysql;
+import ballerina/http;
+import ballerina/runtime;
+import ballerina/observe;
+import ballerina/log;
+
+type Marks record {
+    int student_Id,
+    int maths,
+    int english,
+    int science,
+
+};
+
+endpoint mysql:Client testDB1 {
+    host: "localhost",
+    port: 3306,
+    name: "testdb",
+    username: "root",
+    password: "",
+    poolOptions: { maximumPoolSize: 5 },
+    dbOptions: { useSSL: false }
+};
+
+// This service listener
+endpoint http:Listener listener {
+    port: 9191
+};
+
+// Service for the Student data service
+@http:ServiceConfig {
+    basePath: "/marks"
+}
+service<http:Service> MarksData bind listener {
+    @http:ResourceConfig {
+        methods:["GET"],
+        path: "/getMarks/{stuId}"
+    }
+    // getMarks resource used to get student's marks
+    getMarks(endpoint httpConnection, http:Request request, int stuId) {
+        http:Response response = new;
+        json result = findMarks(untaint stuId);
+
+
+        //Pass the obtained json object to the requested client
+        response.setJsonPayload(untaint result);
+        _ = httpConnection->respond(response);
+    }
+}
+
+documentation {
+  `findMarks` is a function to find a student's marks from the marks record database
+   P{{stuId}} This is the id of the student.
+   R{{}} This function returns a json object. If data is found it returns json containing a table of data which is converted to json format.
+            If data is not added , it returns the json containing a status and error message.
+}
+
+public function findMarks(int stuId) returns (json) {
+    json status = {};
+    io:println("reached");
+    int spanId = check observe:startSpan("Select Data");
+    // Self defined span for observability purpose
+    string sqlString = "SELECT * FROM marks WHERE student_Id = " + stuId;
+    // Getting student marks of the given ID
+    io:println(stuId);
+    var ret = testDB1->select(sqlString, Marks, loadToMemory = true);
+    //Invoking select operation in testDB
+    _ = observe:finishSpan(spanId);
+    // Stopping the previously started span
+
+    //Assigning data obtained from db to a table
+    table<Marks> datatable;
+    match ret {
+        table tableReturned => datatable = tableReturned;
+        error e => {
+            io:println("Select data from student table failed: "
+                    + e.message);
```

Comment:
Shall we log the error instead of println to be consistent in all places?

Reply:
Noted. I will make all the above changes